frontend/src/components/Graph.jsx

```diff
@@ -0,0 +1,158 @@
+import { useEffect, useRef } from 'react';
+import { select, schemeCategory10, forceSimulation, forceLink, forceManyBody, forceX, forceY, scaleOrdinal, drag } from 'd3';
+import { legendColor } from 'd3-svg-legend';
+
+function Graph({ rawDocs, displayedDocs }) {
+  const svgRef = useRef();
+  let width = window.innerWidth;
+  const height = width * 0.7;
+
+  useEffect(() => {
+    let docs = rawDocs || [];
+    const types = Array.from(new Set(docs
+      .flatMap(d => d[2])
+      .filter(x => x !== undefined)
+      .map(x => x.verb)
+    ));
+    const typeColorScale = scaleOrdinal(types, schemeCategory10);
+    const nodes = docs.map(d => {
+      return { id: d[0], title: d[1] };
+    });

```

Comment:
Don't use return when you have a single line function:

```
d => ({ id: d[0], title: d[1] })
```

The enclosing parenthesis is here so that `{}` are interpreted as enclosing an object and not a code block.

Reply:
Or you can even replace it with the more elegant:

```
[id, title] => ({id, title})
```